TestProject/ClientApp/src/components/Posts.js

```diff
@@ -1,11 +1,11 @@
 ﻿import React, { useEffect, useState } from 'react';
 import Post from './Post'
 
-function Posts({ userLoggedIn, userProfile }) {
+function Posts({ userLoggedIn, userProfile, url }) {
```

Comment:
You shouldn't pass `url` as a property. Component itself should know how to display what it is responsible for. You can pass options. Further, you don't need 2 separate API methods to retrieve the same info. Pass options and filter posts by them on the backend.